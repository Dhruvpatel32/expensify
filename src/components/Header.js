import {NavLink} from 'react-router-dom';
import React from 'react';
const Header =()=>(<div>
    <h1>Expensify</h1>
 <NavLink to="/" activeClassName="is-active" exact={true}>home</NavLink>
 <NavLink to="/create" activeClassName="is-active">add</NavLink>
 <NavLink to="/editexpanse" activeClassName="is-active">Edit</NavLink>

</div>
);
export default Header;