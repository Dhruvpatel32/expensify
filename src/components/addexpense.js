import React from 'react';
import ExpanseForm from './expenseForm';
import {connect} from 'react-redux';
import {addExpense} from '../actions/expense'
  export  class AddExpense extends React.Component{
    onSubmit=(expanse)=>{
        this.props.onSubmit(expanse);
        this.props.history.push("/")}
   render(){
       return (
        <div>
        <h1>Add Expanse</h1>
        <ExpanseForm
            onSubmit={this.onSubmit}
            
        />
    </div>
       )
   }     
}

const mapdispatchToProps=(dispatch)=>({
    onSubmit:(expanse)=>dispatch(addExpense(expanse))
})
 export default connect(undefined,mapdispatchToProps)(AddExpense);