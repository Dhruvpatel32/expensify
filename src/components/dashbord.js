import React from 'react';
import ExpenseList from './expenseList';
import ExpenseListFilterText from './expenseListTextFilter';
const ExpanseDashBordpage=()=>(
    <div>
        <ExpenseListFilterText/>
        <ExpenseList/>

    </div>
);
export default ExpanseDashBordpage;