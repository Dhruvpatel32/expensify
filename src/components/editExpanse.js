import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './expenseForm';
import {editExpense} from '../actions/expense';
import {removeById} from '../actions/expense';
export class editExpanse extends React.Component{
onSubmit=(expense)=>{
    this.props.dispatch(editExpense(this.props.expense.id,expense));
    this.props.history.push("/");}
onClick= ()=>{
    this.props.dispatch(removeById({id:this.props.expense.id}));
    this.props.history.push('/')}  
    render(){
        return (<div>
            <ExpenseForm 
        expense={this.props.expense}
        onSubmit={this.onSubmit} />
       <button onClick={this.onClick}>Remove</button>
        </div>) ;
    } 
}



const mapStatetoProps=(state,props)=>{
    const expense=state.expense.find((expenses)=>  expenses.id===props.match.params.id )
    return {
        expense:expense
          
    };
};
export default connect(mapStatetoProps)(editExpanse);