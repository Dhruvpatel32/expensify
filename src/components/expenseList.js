import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './expenseListitems';
import selectExpenses from '../selectors/expenses';


 

export const  ExpenseList=(props)=>(
    <div>
        <h1>Expense list</h1>
       {
        props.expense.length===0 ?
        <p>no expense are avaliable</p>:
        
            props.expense.map((state)=>{
             return <ExpenseListItem key={state.id}{...state}/>
            })
        
       }
        
      
    </div>
)
export const mapStatetoProps=(state)=>{
    return {
       
        expense:selectExpenses(state.expense,state.filter)
        
    };
}
export  default connect(mapStatetoProps)(ExpenseList);
