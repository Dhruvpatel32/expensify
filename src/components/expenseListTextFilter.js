import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter,sortBydate,sortByAmount,setStartDate,setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';
class ExpenseListTextFilter extends React.Component{
    state={
        calendarFocused:null
    };
    onDatesChange=({startDate,endDate})=>{
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }
    onFocusChange=(calendarFocused)=>{
       this.setState(()=>({calendarFocused})); 
    }
    render(){
        return (
            <div className="container">
            <div className="container_main">
            <div className="list_items">
            <input className="text-input" placeholder="Serch Expenses" type="text" value={this.props.filter.text} onChange={(e)=>{
              {this.props.dispatch(setTextFilter(e.target.value))}
            }}/>
            </div>
            <div className="list_items">
            <select className="select" vlaue={this.props.filter.sortBy} onChange={(e)=>{
                if(e.target.value==='date'){
                    this.props.dispatch(sortBydate())
                }
                else if(e.target.value==='amount'){
                    this.props.dispatch(sortByAmount())
                }
            }}>
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>
            </div>
            <div className="list_items">
            <DateRangePicker
               startDate={this.props.filter.startDate}
               endDate={this.props.filter.endDate}
               onDatesChange={this.onDatesChange}
               focusedInput={this.state.calendarFocused}
               onFocusChange={this.onFocusChange}
               numberOfMonths={1}
               showClearDates={true}
               isOutsideRange={()=>false}

           />
            </div> 
        </div>
        </div>
        )
    } 
}
const mapStateToProps=(state)=>{
    return {
        filter:state.filter
    }
}

export default connect(mapStateToProps)(ExpenseListTextFilter)