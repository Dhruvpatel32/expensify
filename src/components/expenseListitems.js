import React from 'react';


import {Link} from 'react-router-dom';


 

const Expenselist=({id,description,amount,createdAt})=>(
    <div>
        <Link to={`/editexpanse/${id}`}>
        <h3>{description}</h3>
        </Link>
        <p>{amount} ----{createdAt}</p>
        
    </div>
);

export default (Expenselist);