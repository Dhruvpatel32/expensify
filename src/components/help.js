import React from 'react';
const help=()=>(
    <div>
        <h1>This is help page!!!</h1>
    </div>
);
export default help; 