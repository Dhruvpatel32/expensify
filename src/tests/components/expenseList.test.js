import React from 'react';
import {ExpenseList} from '../../components/expenseList';
import {shallow} from 'enzyme';
import moment from 'moment';
import Adapter from 'enzyme-adapter-react-16';
const expense=[
    {
        id:"1",
        description:"gum",
        amount:195,
        note:"",
        createdAt:0
    },
   {
        id:"2",
        description:"rent",
        amount:19500,
        note:"",
        createdAt:moment(0).subtract(4,'day').valueOf()
    },
    {
        id:"3",
        description:"credit cacrd",
        amount:495,
        note:"",
        createdAt:moment(0).add(4,'day').valueOf()
    }
];
test('should render expenseList',()=>{
    const wrapper=shallow(<ExpenseList expense={expense}/>)
    expect(wrapper).toMatchSnapshot();
})
test('should render expenseList when there is no expense',()=>{
    const wrapper=shallow(<ExpenseList expense={[]}/>)
    expect(wrapper).toMatchSnapshot();
})
