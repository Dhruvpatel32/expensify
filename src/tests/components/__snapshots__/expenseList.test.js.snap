// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseList 1`] = `
<div>
  <h1>
    Expense list
  </h1>
  <Expenselist
    amount={195}
    createdAt={0}
    description="gum"
    id="1"
    key="1"
    note=""
  />
  <Expenselist
    amount={19500}
    createdAt={-345600000}
    description="rent"
    id="2"
    key="2"
    note=""
  />
  <Expenselist
    amount={495}
    createdAt={345600000}
    description="credit cacrd"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`should render expenseList when there is no expense 1`] = `
<div>
  <h1>
    Expense list
  </h1>
  <p>
    no expense are avaliable
  </p>
</div>
`;
